jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/RetrySyncWorkItemTest.java

```diff
@@ -0,0 +1,108 @@
+package org.jbpm.test.functional.workitem;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.drools.core.process.instance.WorkItemManager;
+import org.jbpm.test.JbpmTestCase;
+import org.jbpm.workflow.instance.WorkflowRuntimeException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.process.NodeInstance;
+//import org.jbpm.workflow.instance.NodeInstance;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.api.task.TaskService;
+import org.kie.api.task.model.TaskSummary;
+import org.kie.internal.runtime.manager.context.EmptyContext;
+
+@RunWith(Parameterized.class)
+public class RetrySyncWorkItemTest extends JbpmTestCase {
+
+    private static final String RETRY_WORKITEM_JPA_PROCESS_ID = "org.jbpm.test.retryWorkitem-jpa";
+    private static final String RETRY_WORKITEM_MEMORY_PROCESS_ID = "org.jbpm.test.retry-workitem-memory";
+    private KieSession kieSession;
+    private WorkItemManager workItemManager;
+    private RuntimeEngine runtimeEngine;
+
+    @Parameters
+    public static Collection<Boolean[]> init() {
+        return Arrays.asList( new Boolean[][]{{false, false}, {true, true}} );
+    }
+
+    public RetrySyncWorkItemTest(Boolean session, Boolean dataSource) {
+        super( session, dataSource );
+    }
+
+    @Before
+    public void setup() {
+        manager = createRuntimeManager( "org/jbpm/test/functional/workitem/retry-workitem-jpa.bpmn2", "org/jbpm/test/functional/workitem/retry-workitem-memory.bpmn2" );
+        runtimeEngine = getRuntimeEngine( EmptyContext.get() );
+        kieSession = runtimeEngine.getKieSession();
+        workItemManager = (org.drools.core.process.instance.WorkItemManager) kieSession.getWorkItemManager();
+    }
+
+    private void workItemRecoveryTestByJPA() {
+        workItemManager.registerWorkItemHandler( "ExceptionWorkitem", new ExceptionWorkItemHandler() );
+        ProcessInstance pi = kieSession.startProcess( RETRY_WORKITEM_JPA_PROCESS_ID );
+        TaskService taskService = runtimeEngine.getTaskService();
+
+        assertProcessInstanceActive( pi.getId() );
+        assertNodeTriggered( pi.getId(), "lockingNode" );
+        assertNodeActive( pi.getId(), kieSession, "lockingNode" );
+
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsPotentialOwner( "john", "en-UK" );
+        assertEquals( 1, tasks.size() );
+        TaskSummary taskSummary = tasks.get( 0 );
+
+        taskService.start( taskSummary.getId(), "john" );
```

Comment:
yes, i use two process definitions, because of i cann't get TaskService taskService = runtimeEngine.getTaskService() in memory model. so cann't handle humantask.

I will refacing it to two unit test.